doc/nrf/app_dev/device_guides/nrf54h/ug_nrf54h20_suit_recovery.rst

```diff
@@ -236,3 +236,177 @@ To turn an application into a recovery application, the following steps have to
+   This can be done by adding ``-Drecovery_CONFIG_SUIT_INVOKE_APP_LOCAL_3_BEFORE_MAIN_APP=y`` to the build command when building the recovery application.
+
+   However, otherwise, the manifest templates have to be modified as follows:
+
+a. In the APP_LOCAL_3 manifest template, add the following code before the ``suit-directive-invoke`` directive on the companion image:
+
+   .. code-block:: yaml
+
+      - suit-directive-override-parameters:
+          suit-parameter-invoke-args:
+            suit-synchronous-invoke: True
+            suit-timeout: 1000
+
+b. In the root manifest template add the ``INSTLD_MFST`` component for the local manifest orchestrating the comapanion image.
+   Make sure it is invoked before the main application images.
+   If the modified template is based on the root manifest template from |NCS|, it will in most cases be enough to add it as the first manifst in the component list.
+   The following code snippet is responsible for this in the default root manifest template:
+
+   .. code-block:: yaml
+
+      {{- component_list.append( app_recovery_local_component_index ) or ""}}
+      - - INSTLD_MFST
+        - RFC4122_UUID:
+            namespace: {{ mpi_app_recovery_local_vendor_name }}
+            name: {{ mpi_app_recovery_local_class_name }}
+
+
+.. _ug_nrf54h20_suit_recovery_enter_via_button:
+
+Entering recovery via button press
+==================================
+
+A common use case for entering recovery mode is to allow the device to enter recovery mode by pressing a button.
+In NCS this use case is provided as a simple plug-and-play solution.
+When it is enabled, the device will enter recovery mode if the button is pressed during boot.
+
+.. note::
+   If no modifications or additional modules are added to the recovery application the only way to exit the recovery mode is performing a device firmware upgrade.
+
+There are two variants of this feature:
+
+* Recovery button checked in the recovery application run as a companion image at boot stage (recommended).
+* Recovery button checked in the main application.
+
+The first variant is recommended as it allows to perform the check independently of the main application.
+
+Recovery button checked in the recovery application
+---------------------------------------------------
+
+In this variant,the recovery application is run as a companion image before the main application as part of the normal invoke path.
+It checks if the device already is in recovery mode.
+If it is, the recovery application proceeds with its normal operation allowing to recover firmware.
+
+If the device is not in recovery mode, the recovery application checks if the specified button is pressed.
+If the button is pressed, the recovery application sets the recovery flag and reboots the device.
+Otherwise it sends a confirmation message to the Secure Domain Firmware, which then halts the recovery application and proceeds with booting the main application.
```

Comment:
```suggestion
Otherwise, it sends a confirmation message to the Secure Domain Firmware, which then halts the recovery application and proceeds with booting the main application.
```